scripts/class/Hero.js

```diff
@@ -5,6 +5,20 @@ class Hero {
     */
     constructor(level){
         //TODO generate initial hero/random monster with random stats
+        this.name = '';
+        this.experience = 0;
+        this.level = level;
+        this.stats = new Stats();
+        this.avatar = ''; //URI to hero image
+        this.position = new Position();
+        this.gear = {
+            left: new Item(),
+            right: new Item(),
+            chest: new Item(),
+            head: new Item()
+        };
+        this.carry = new Array(4); // Item
```

Comment:
No need. `this.carry = []` for a new Hero is enough.
However for an existing hero from json, we should iterate through its items and pass them to the item constructor before filling them in the carry array. Like:
`this.carry = hero.carry.map(item => new Item(item));` This will also create an empty array if no items are in carry I believe.